tpm.h

```diff
@@ -25,23 +25,29 @@
 #define TPM_NO_LOCALITY		0xFF
 
 enum tpm_hw_intf {
-	TPM_DEVNODE,
 	TPM_TIS,
-	TPM_CRB,
-	TPM_UEFI
+	TPM_CRB
 };
 
 enum tpm_family {
 	TPM12,
 	TPM20
 };
 
+struct tpm_hw_ops {
+	u8 request_locality(u8);
+	void relinquish_locality(void);
+	size_t send(struct tpmbuff *);
+	size_t recv(enum tpm_family, struct tpmbuff *);
```

Comment:
crb_recv() does not match this function pointer signature. You need to add family.

Reply:
ack